Makefile

```diff
@@ -0,0 +1,28 @@
+default: help
+
+install-env: ## Install conda environement
+	conda env create -f binder/environment.yml
+.PHONY: install-env
+
+
+update-env: init ## Update conda environement
+	conda env update -f binder/environment-dev.yml
+.PHONY: update-env
+
+
+tests: ## Run tests
+	pytest tests
+.PHONY: tests
+
+
+lint: ## Lint code
+	pycodestyle buildh \
+	&& pydocstyle buildh \
+	&& pylint buildh
+.PHONY: lint
+
+
+help:
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```

Comment:
I don't understand that one. I'm not very familiar with Makefile, what does `$(MAKEFILE_LIST)` contains?

Reply:
`$(MAKEFILE_LIST)` corresponds to the [name](https://ftp.gnu.org/old-gnu/Manuals/make-3.80/html_node/make_17.html) of the Makefile itself.

`help` is the default rule for the Makefile and produces a summary of available rules:
```
$ make
install-env                    Install conda environement
lint                           Lint code
tests                          Run tests
update-env                     Update conda environement
```

Reply:
Thanks 👍